overlord/hookstate/ctlcmd/helpers.go

```diff
@@ -228,11 +228,78 @@ func queueDefaultConfigureHookCommand(context *hookstate.Context, tts []*state.T
 	return nil
 }
 
+func maybePatchServiceNames(snapInstance string, serviceNames []string) (
+	updatedServiceNames []string, patched bool, err error,
+) {
+	snapName, snapInstanceKey := snap.SplitInstanceName(snapInstance)
+	hasInstanceKey := snapInstanceKey != ""
+
+	if !hasInstanceKey {
+		// no patching needed, return names as they are
+		return serviceNames, false, nil
+	}
+
+	// Backward compatibility path for a scenario when snapctl service operation
+	// is called in a context of a snap with instance key. It is possible that
+	// the request uses service names of form 'snap.app' which does not include
+	// an instance key. We want to 'patch' them to 'snap_foo.app', such that
+	// existing snaps that aren't completely aware of parallel installs work
+	// correctly.
+
+	updatedServiceNames = make([]string, 0, len(serviceNames))
+	keyedCnt := 0
```

Comment:
can you add a comment to define what this is, I had to think a moment to understand is counting when instance key appears. Or maybe call this withInstanceKeyCnt ?